docs/rules/jsx-no-undef-props.md

```diff
@@ -0,0 +1,23 @@
+# Disallow prop value as `undefined`. (jsx-no-undef-props)
+
+This rule prevents setting a prop value as `undefined` and suggests you just don't pass the prop down. In React, that prop will be dropped by the parent, anyway.
+
+## Rule Details
+
+The following patterns are considered warnings:
+
+```js
+<Hello name={undefined} />;
+<Hello name={foo ? undefined : bar} />;
```

Comment:
True - you can move the logic outside of the prop value and just pass a var, but should probably just special case this to ignore. In this case, the rule is kinda pointless :)